src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1366,11 +1366,39 @@ MavlinkReceiver::handle_message_set_attitude_target(mavlink_message_t *msg)
 						att_sp.yaw_sp_move_rate = 0.0f;
 					}
 
-					// TODO: We assume offboard is only used for multicopters which produce thrust along the
-					// body z axis. If we want to support fixed wing as well we need to handle it differently here, e.g.
-					// in that case we should assign att_sp.thrust_body[0]
 					if (!_offboard_control_mode.ignore_thrust) { // dont't overwrite thrust if it's invalid
-						att_sp.thrust_body[2] = -set_attitude_target.thrust;
+						// Fill correct field by checking frametype
+						// TODO: add as needed
+						switch (_mavlink->get_system_type()) {
+						case MAV_TYPE_GENERIC:
+							break;
+
+						case MAV_TYPE_FIXED_WING:
```

Comment:
Hm, so what happens for fixedwing?

Reply:
> 
> 
> Hm, so what happens for fixedwing?

I don' t know. Maybe we need someone focusing on fixed-wings?

Reply:
> Hm, so what happens for fixedwing?

Attitude control from offboard mode makes no sense on a fixed-wing in my opinion. It should focus on velocity and position control instead. And it seems its controller only deals with `att_sp.thrust_body[0]` but ask for all of them.